TACC/Core-CMS-Custom
# hotfix/WP-232-234-APCD-frontend-form-sanitation

## Overview
The purpose of this hotfix is to still allow special characters that users may use but restricting invalid characters on form submissions. 
Your thoroughness of testing is really appreciated! I know it's tedious. Using Django or React utilities in place of this band-aid is the long term goal once the fire is put out.


## Related

- [WP-232](https://jira.tacc.utexas.edu/browse/WP-232)
- [WP-233](https://jira.tacc.utexas.edu/browse/WP-233)
- [WP-234](https://jira.tacc.utexas.edu/browse/WP-234)
- mimicked by #201

## Changes

The checkForBlankInputs javascript was changed using whitelist methods to allow characters in input. At the same time, the script blacklists some explicit scripting character patterns. This is a front end change to prevent bad characters from making it to the server. The DB utils remove bad characters server side in apcd_database.py. 


## Testing

1. Go to [http://localhost:8000/register/request-to-submit/](http://localhost:8000/register/request-to-submit/)
2. Try to input **<script> </script> javascript.exe, anything in "quotes" or 'quotes" { hello.exe } thisshouldwork@test.com into any field** into a field
3. On change. the field should be replaced  with **Try to input script script exe anything in quotes or quotes  helloexe  thisshouldworktestcom**
4. Go to [http://localhost:8000/submissions/threshold-exception/](http://localhost:8000/submissions/threshold-exception/)
5. Repeat steps 2 and 3
6. Go to [http://localhost:8000/submissions/other-exception/](http://localhost:8000/submissions/other-exception/)
7. Repeat steps 2 and 3
8. Go to [https://apcd-qa.tacc.utexas.edu/submissions/extension-request/](https://apcd-qa.tacc.utexas.edu/submissions/extension-request/)
9.  Repeat steps 2 and 3


## UI

###User input:

<img width="1434" alt="Screenshot 2023-08-22 at 6 07 51 PM" src="https://github.com/TACC/Core-CMS-Custom/assets/96220951/ae9b3058-e23f-43ca-b125-40f148f15a00">

###On change, text is changed to :

<img width="988" alt="Screenshot 2023-08-22 at 6 08 53 PM" src="https://github.com/TACC/Core-CMS-Custom/assets/96220951/7cac8006-4364-42dc-b8de-1bce45fc27d9">
